Tests/DatadogTests/Datadog/Utils/SwiftExtensionsTests.swift

```diff
@@ -0,0 +1,21 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import XCTest
+@testable import Datadog
+
+class DateExtensionTests: XCTestCase {
+    func testCurrentMillis() {
+        let date15Dec2019 = Date.mockDecember15th2019At10AMUTC()
+        XCTAssertEqual(date15Dec2019.currentTimeMillis, 1_576_404_000_000)
+
+        let dateAdvanced = date15Dec2019 + 9.999
+        XCTAssertEqual(dateAdvanced.currentTimeMillis, 1_576_404_009_999)
+
+        let dateAgo = date15Dec2019 - 0.001
+        XCTAssertEqual(dateAgo.currentTimeMillis, 1_576_403_999_999)
+    }
```

Comment:
Let's also include test for the code branch which captures overflow, this can be easily done by using "reference date" as relative base:

```suggestion
    }

    func testUInt64Overflow() {
        let dateWithOverflow = Date(timeIntervalSinceReferenceDate: .greatestFiniteMagnitude)
        XCTAssertEqual(dateWithOverflow.currentTimeMillis, UInt64.max)
    }
```

Reply:
good idea 💡 
i just noticed i forgot to add tests for `init(withReportingOverflow:)` too